prov/tcp/src/tcpx_ep.c

```diff
@@ -421,6 +421,55 @@ void tcpx_rx_entry_free(struct tcpx_xfer_entry *rx_entry)
 	}
 }
 
+/* Must hold ep->lock. */
+static void tcpx_ep_cancel_rx(struct tcpx_ep *ep, void *context)
+{
+	struct slist_entry *cur, *prev;
+	struct tcpx_xfer_entry *xfer_entry;
+	struct tcpx_cq *cq;
+
+	/* To cancel an active receive, we would need to flush the socket of
+	 * all data associated with that message.  Since some of that data
+	 * may not have arrived yet, this would require additional state
+	 * tracking and complexity.  Fail the cancel in this case, since
+	 * the receive is already in process anyway.
+	 */
+	slist_foreach(&ep->rx_queue, cur, prev) {
+		xfer_entry = container_of(cur, struct tcpx_xfer_entry, entry);
+		if (xfer_entry->context == context) {
+			if (ep->cur_rx_entry == xfer_entry)
```

Comment:
@shefty I recently looked at the cancelation code again and the following condition seems incorrect to me. Aren't we supposed to fail the cancelation if `ep->cur_rx_entry == xfer_entry`, like the comment states?

I would expect the code to be as follows:
```
if (ep->cur_rx_entry != xfer_entry)
    goto found;
break;```

Reply:
uhm... yeah, the check looks backwards